packages/blade/src/components/Checkbox/Checkbox.tsx

```diff
@@ -97,21 +99,25 @@ type CheckboxProps = {
   size?: 'small' | 'medium';
 };
 
-const Checkbox = ({
-  defaultChecked,
-  validationState,
-  isChecked,
-  isDisabled,
-  isIndeterminate,
-  isRequired,
-  name,
-  onChange,
-  value,
-  children,
-  helpText,
-  errorText,
-  size = 'medium',
-}: CheckboxProps): React.ReactElement => {
+const _Checkbox: React.ForwardRefRenderFunction<BladeRefElement, CheckboxProps> = (
+  {
+    defaultChecked,
+    validationState,
+    isChecked,
+    isDisabled,
+    isIndeterminate,
+    isRequired,
+    name,
+    onChange,
+    value,
+    children,
+    helpText,
+    errorText,
+    size = 'medium',
+  },
+  ref,
+) => {
+  const inputRef = useBladeInnerRef(ref);
```

Comment:
what if we move this to SelectorInput then all we'll have to do is forward the ref arg and then handle it inside SelectorInput so that way even the radio will get it?

Reply:
I'll try it out and see if it could work.

Reply:
Works, though it doesn't simplify the code, just moves the useBladeInnerRef to SelectorInput. 

I'm fine with either of them. will push the change.